development/webpack/utils/plugins/ManifestPlugin/helpers.ts

```diff
@@ -29,28 +38,72 @@ export function transformManifest(args: { lockdown: boolean; test: boolean }) {
     transforms.push(removeLockdown);
   }
 
-  function addTabsPermission(browserManifest: chrome.runtime.Manifest) {
-    if (browserManifest.permissions) {
-      if (browserManifest.permissions.includes('tabs')) {
-        throw new Error(
-          "manifest contains 'tabs' already; this transform should be removed.",
+  /**
+   * This function sets predefined flags in the manifest's _flags property
+   * that are stored in the .manifest-flags.json file.
+   *
+   * @param browserManifest - The Chrome extension manifest object to modify
+   */
+  function addManifestFlags(browserManifest: chrome.runtime.Manifest): void {
+    let manifestFlags;
+
+    if (manifestOverridesPath) {
+      try {
+        const fs = require('node:fs');
+        const path = require('node:path');
+        const manifestFlagsContent = fs.readFileSync(
+          path.resolve(process.cwd(), manifestOverridesPath),
+          'utf8',
         );
+        manifestFlags = JSON.parse(manifestFlagsContent);
+      } catch (error: unknown) {
+        if (
+          error instanceof Error &&
+          'code' in error &&
+          error.code === 'ENOENT'
+        ) {
+          // Only throw if ENOENT and manifestOverridesPath was provided
+          throw new Error(
+            `Manifest override file not found: ${manifestOverridesPath}`,
+          );
+        }
       }
+    }
+
+    if (manifestFlags) {
+      merge(browserManifest, manifestFlags);
+    }
+  }
+
+  if (isDevelopment) {
+    // Add manifest flags only for development builds
+    transforms.push(addManifestFlags);
+  }
+
+  function addTabsPermission(browserManifest: chrome.runtime.Manifest) {
+    if (browserManifest.permissions?.includes('tabs')) {
+      throw new Error(
+        "manifest contains 'tabs' already; this transform should be removed.",
+      );
+    }
+    if (browserManifest.permissions) {
       browserManifest.permissions.push('tabs');
     } else {
       browserManifest.permissions = ['tabs'];
     }
   }
+
   if (args.test) {
     // test builds need "tabs" permission for switchToWindowWithTitle
     transforms.push(addTabsPermission);
   }
 
   return transforms.length
     ? (browserManifest: chrome.runtime.Manifest, _browser: string) => {
-        const clone = structuredClone(browserManifest);
-        transforms.forEach((transform) => transform(clone));
-        return clone;
+        // Create a deep copy of the manifest to avoid modifying the original
+        const manifestCopy = JSON.parse(JSON.stringify(browserManifest));
```

Comment:
Why the change from `structuredClone` to `JSON.parse(JSON.stringify(browserManifest));`. AFAIK, `structuredClone` was made for this kind of situation.

docs: https://developer.mozilla.org/en-US/docs/Web/API/Window/structuredClone

Reply:
My bad, it is a leftover from refactor, modified 49542c0edd4caa5c7d8f0394c39676d74d50c554